README.md

```diff
@@ -1,54 +1,141 @@
-    - `api/core/lib/x64/fmod.dll`
-    - `api/core/lib/x64/fmod_vc.lib`
-    - `api/studio/lib/x64/fmodstudio.dll`
-    - `api/studio/lib/x64/fmodstudio_vc.lib`
+The linking process on Windows is straight forward. Download and install the
+"FMOD Engine" for Windows. When installing, a folder will be created with FMOD
+libraries. Copy the followin g files into root of your project:
+
+> [!WARNING]
+> TODO: Update this section with the correct file names.
+
+When publishing your game, you will need to include the non-debugging versions
+of the libraries in the same directory as the executable.
+
+```
+My Game/
+├── My Game.exe
+├── fmod.dll
+└── fmodstudio.dll
+```
+
+### MacOS
+
+- Download "FMOD Engine" for MacOS.
+- In the dmg file, open the `FMOD Programmers API` folder.
+- You will need these files:
+    - `api/core/lib/libfmod.dylib`
+    - `api/core/lib/libfmodL.dylib` (debugging only)
+        - `api/studio/lib/libfmodstudio.dylib`
+    - `api/studio/lib/libfmodstudioL.dylib` (debugging only)
+
+Linking on MacOS is a bit different to Windows, as the defaults of the OS are
+not as straight forward as Windows. Also, Windows seems to take parent
+directories into account. During development, it is sufficient to put the
+libraries in the root of your project. When building, the built executable is
+contained in the `target/debug` directory. Now on Windows, this doesn't seem to
+be a problem, but on MacOS, the executable is unable to find the libraries.
+
+To fix this, you have to pass some flags to cargo during development. Have a
+look at this `.cargo/config.toml`:
+
+```toml
+[target.aarch64-apple-darwin]
+rustflags = [
+    "-L", "native=./vendor/fmod",
+    "-C", "link-arg=-Wl,-rpath,./vendor/fmod",
+]
+```
+
+The first line tells cargo where to look for libraries during development. If
+you keep the libraries in the root of your project, there is no problem when
+building the project. However, when running the executable, it will not find the
+libraries, as the executable is in the `target/debug` directory.
+
+I recommend putting the libraries in a known folder like `vendor/fmod` and then
+pass the path to cargo using the
+`-L` flag. There are different ways to do this, but using the
+`.cargo/config.toml` is the most convenient to me.
+
+The second line will determine the rpath of the executable. This is the path
+where the executable will look for the libraries. By default, executables will
+look in a variety of places, including the directory the executable is in. This
+is fine when publishing the game, as you can just use the Windows method and put
+the libraries in the same directory as the executable. However, during
+development, the executable is in the `target/debug` directory, which gets
+generated automatically by cargo and does not contain the libraries. The
+`"-C", "link-arg=-Wl,-rpath,./vendor/macos"` flag will tell the executable to
+look in the `vendor/fmod` directory of your project for the libraries.
```

Comment:
The path mentioned is relative to the project root, which the executable doesn't know about. So I assume you mean to say that it will tell the executable to look there when using `cargo run`. Might be good to mention that (not sure how to do it succinctly though :cry: ).

Reply:
This information is for development only. When the user runs the program from an IDE, the IDE will set the working directory to the folder of the project and then execute the binary. That way the executable will be able to resolve the library path.